iotdb-core/datanode/src/main/java/org/apache/iotdb/db/schemaengine/schemaregion/mtree/impl/pbtree/cache/ReleaseFlushMonitor.java

```diff
@@ -0,0 +1,350 @@
+        regionAndFreeTimeList.add(new Pair<>(regionId, traverserFreeTime));
+      }
+    }
+    regionAndFreeTimeList.sort(Comparator.comparing((Pair<Integer, Long> o) -> o.right).reversed());
+    return regionAndFreeTimeList.stream().map(Pair::getLeft).collect(Collectors.toList());
+  }
+
+  @TestOnly
+  public void forceFlushAndRelease() {
+    scheduler.forceFlushAll();
+    scheduler.scheduleRelease(true);
+  }
+
+  public void clear() {
+    if (flushMonitor != null) {
+      flushMonitor.shutdownNow();
+      while (true) {
+        if (flushMonitor.isTerminated()) break;
+      }
+      flushMonitor = null;
+    }
+    if (scheduler != null) {
+      scheduler.clear();
+      while (true) {
+        if (scheduler.isTerminated()) break;
+      }
+      scheduler = null;
+    }
+    regionToStoreMap.clear();
+    flushingRegionSet.clear();
+    regionToTraverserTime.clear();
+    releaseFlushStrategy = null;
+    engineStatistics = null;
+    releaseSemaphore = null;
+    engineMetric = null;
+  }
+
+  public int getActiveWorkerNum() {
+    return scheduler.getActiveWorkerNum();
+  }
+
+  private ReleaseFlushMonitor() {}
+
+  private static class GlobalCacheManagerHolder {
+    private static final ReleaseFlushMonitor INSTANCE = new ReleaseFlushMonitor();
+
+    private GlobalCacheManagerHolder() {}
+  }
+
+  public static ReleaseFlushMonitor getInstance() {
+    return ReleaseFlushMonitor.GlobalCacheManagerHolder.INSTANCE;
+  }
+
+  private static class RecordList {
+    // The start time of RecordNode is incremental from head to tail
+    private final RecordNode head = new RecordNode();
+    private final RecordNode tail = new RecordNode();
+
+    private RecordList() {
+      head.next = tail;
+      tail.prev = head;
+    }
+
+    private synchronized RecordNode createAndAddToTail() {
+      RecordNode recordNode = new RecordNode();
+      recordNode.prev = tail.prev;
+      recordNode.next = tail;
+      tail.prev.next = recordNode;
+      tail.prev = recordNode;
+      return recordNode;
+    }
+
+    private synchronized void remove(RecordNode recordNode) {
+      recordNode.prev.next = recordNode.next;
+      recordNode.next.prev = recordNode.prev;
+      recordNode.prev = null;
+      recordNode.next = null;
+    }
+
+    private synchronized void clear() {
+      head.next = tail;
+      tail.prev = head;
+    }
+
+    private Iterator<RecordNode> iterator() {
+      return new Iterator<RecordNode>() {
+        private RecordNode cur = head;
+
+        @Override
+        public boolean hasNext() {
+          return cur.next != tail;
+        }
+
+        @Override
+        public RecordNode next() {
+          if (!hasNext()) {
+            throw new NoSuchElementException();
+          }
+          cur = cur.next;
+          return cur;
+        }
+
+        @Override
+        public void remove() {
+          RecordList.this.remove(cur);
+        }
```

Comment:
Might cause NPE if the cur is removed during iterating.

Reply:
Actually there are only one thread using this iterator... NPE will not be occurred without concurrency. So is it ok to add @NotThreadSafe?

Reply:
I will correct my description. After removal, the iterator is bound to return null. Is this expected?

Reply:
Fixed